Add env var for auto_multi_line_detection (#13054)
DataDog/documentation: content/en/agent/logs/advanced_log_collection.md

  auto_multi_line_detection: true
```

> For containerized deployments this can be enabled with the environment variable `DD_LOGS_CONFIG_AUTO_MULTI_LINE_DETECTION=true`.

Reviewer: ```suggestion
For containerized deployments, you can enable `auto_multi_line_detection` with the `DD_LOGS_CONFIG_AUTO_MULTI_LINE_DETECTION=true` environment variable.
```
